File: 07-loops/main.go
package main

import "fmt"

func main() {
	// === 基本 for 循环（当 while 用） ===
	i := 0
	for i < 5 {
		fmt.Printf("%d ", i)
		i++
	}
	fmt.Println()

	// === C 风格 for 循环 ===
	sum := 0
	for j := 0; j < 10; j++ {
		if j%2 == 0 {
			continue
		}
		sum += j
	}
	fmt.Printf("奇数之和 (1..9) = %d\n", sum)

	// === range 迭代切片 ===
	fruits := []string{"apple", "banana", "cherry"}
	for _, fruit := range fruits {
		fmt.Printf("%s ", fruit)
	}
	fmt.Println()

	// === range 带索引 ===
	for idx, fruit := range fruits {
		fmt.Printf("[%d] %s\n", idx, fruit)
	}

	// === range 迭代范围（Go 1.22+） ===
	fmt.Print("0 到 4: ")
	for n := range 5 {
		fmt.Printf("%d ", n)
	}
	fmt.Println()

	// === 同时迭代多个切片（Go 不直接支持，需手动索引） ===
	names := []string{"Alice", "Bob", "Carol"}
	ages := []int{30, 25, 28}
	for k := range names {
		fmt.Printf("%s 的年龄是 %d\n", names[k], ages[k])
	}

	// === Go 没有 break 返回值，需要额外变量 ===
	data := []int{1, 3, 7, 4, 9, 2}
	found := -1
	for idx, v := range data {
		if v > 5 {
			found = idx
			break
		}
	}
	if found >= 0 {
		fmt.Printf("第一个大于 5 的元素在索引 %d, 值为 %d\n", found, data[found])
	}

	// === 标签循环 ===
	fmt.Println("乘法表中大于 20 的第一个结果:")
	ra, rb := 0, 0
outer:
	for a := 1; a < 10; a++ {
		for b := 1; b < 10; b++ {
			if a*b > 20 {
				ra, rb = a, b
				break outer
			}
		}
	}
	fmt.Printf("%d x %d = %d\n", ra, rb, ra*rb)

	// === 无限循环 ===
	count := 0
	for {
		count++
		if count >= 3 {
			break
		}
	}
	fmt.Printf("无限循环执行了 %d 次\n", count)
}

File: 07-loops/main.rs
fn main() {
    // === 基本 while 循环 ===
    let mut i: u32 = 0;
    while i < 5 {
        print!("{} ", i);
        i += 1;
    }
    println!();

    // === for..in 迭代范围 ===
    let mut sum: u32 = 0;
    for j in 0..10 {
        if j % 2 == 0 {
            continue;
        }
        sum += j;
    }
    println!("奇数之和 (1..9) = {}", sum);

    // === 迭代切片 ===
    let fruits = ["apple", "banana", "cherry"];
    for fruit in &fruits {
        print!("{} ", fruit);
    }
    println!();

    // === 带索引迭代 ===
    for (idx, fruit) in fruits.iter().enumerate() {
        println!("[{}] {}", idx, fruit);
    }

    // === 范围迭代 ===
    print!("0 到 4: ");
    for n in 0..5 {
        print!("{} ", n);
    }
    println!();

    // === 同时迭代多个切片（用 zip） ===
    let names = ["Alice", "Bob", "Carol"];
    let ages = [30, 25, 28];
    for (name, age) in names.iter().zip(ages.iter()) {
        println!("{} 的年龄是 {}", name, age);
    }

    // === loop + break 返回值 ===
    let data = [1, 3, 7, 4, 9, 2];
    let found = 'search: {
        for (idx, &v) in data.iter().enumerate() {
            if v > 5 {
                break 'search Some(idx);
            }
        }
        None
    };
    if let Some(fi) = found {
        println!("第一个大于 5 的元素在索引 {}, 值为 {}", fi, data[fi]);
    }

    // === 标签循环 ===
    println!("乘法表中大于 20 的第一个结果:");
    let mut ra = 0;
    let mut rb = 0;
    'outer: for a in 1..10 {
        for b in 1..10 {
            if a * b > 20 {
                ra = a;
                rb = b;
                break 'outer;
            }
        }
    }
    println!("{} x {} = {}", ra, rb, ra * rb);

    // === loop 无限循环 ===
    let mut count = 0;
    loop {
        count += 1;
        if count >= 3 {
            break;
        }
    }
    println!("无限循环执行了 {} 次", count);

    // === loop break 返回值（Rust 特色） ===
    let mut c = 0;
    let result = loop {
        c += 1;
        if c == 10 {
            break c * 2; // loop 返回 20
        }
    };
    println!("loop 返回值: {}", result);
}
